### PR TITLE
[Feature] Add Ascend support for RoIAlign op

### PR DESCRIPTION
**Pull request by momo609**
## Motivation

add roi_align op.

## Modification

add roi_align op.

### CODE REVIEW

**Review by ckirchhoff2021 — COMMENTED** [review]
Add unit test.

**Review by ckirchhoff2021 — APPROVED** [review]
/LGTM

File: mmcv/ops/csrc/pytorch/npu/roi_align_npu.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#include "pytorch_npu_helper.hpp"
+
+using namespace NPU_NAME_SPACE;
+using namespace std;
+
+void roi_align_forward_npu(Tensor input, Tensor rois, Tensor output,
+                           Tensor argmax_y, Tensor argmax_x, int aligned_height,
+                           int aligned_width, float spatial_scale,
+                           int sampling_ratio, int pool_mode, bool aligned) {
```
Comment (grimoire) [inline mmcv/ops/csrc/pytorch/npu/roi_align_npu.cpp:9]: Can we add a warning if aligned is False?

File: tests/test_ops/test_roi_align.py [hunk 2]
```diff
@@ -102,15 +102,19 @@ def _test_roialign_allclose(device, dtype):
...
 ])
 @pytest.mark.parametrize('dtype', [
     torch.float,
     pytest.param(
         torch.double,
         marks=pytest.mark.skipif(
-            IS_MLU_AVAILABLE,
-            reason='MLU does not support for 64-bit floating point')),
+            IS_MLU_AVAILABLE or IS_NPU_AVAILABLE,
+            reason='MLU and NPU does not support for 64-bit floating point')),
```
Comment (zhouzaida) [inline tests/test_ops/test_roi_align.py:117]:
```suggestion
            reason='MLU and NPU do not support for 64-bit floating point')),
```

**Pull request merged**